src/apps/intel/intel10g.lua

```diff
@@ -23,7 +23,18 @@ local timer = require("core.timer")
 local bits, bitset = lib.bits, lib.bitset
 local band, bor, lshift = bit.band, bit.bor, bit.lshift
 
-num_descriptors = 1024
+local num_descriptors = 1024
+function ring_buffer_size (arg)
```

Comment:
I'm not sure. The reason I think this should be per module and not per instance is that `num_descriptors` is used to set RDLEN and TDLEN registers in the intel10g driver. In SF that should be OK, but what would happen in VF mode? VF inherits `set_receive_descriptors` and `set_transmit_descriptors` from SF. If two VF instances are started with different `num_descriptors` sizes the latter would override RDLEN and TDLEN registers in the card. Wouldn't that be a problem?